pkg/issuer/vault/setup.go

```diff
@@ -47,7 +47,8 @@ func (v *Vault) Setup(ctx context.Context) error {
 
 	if v.issuer.GetSpec().Vault.Auth.TokenSecretRef.Name != "" &&
 		(v.issuer.GetSpec().Vault.Auth.AppRole.RoleId != "" ||
-			v.issuer.GetSpec().Vault.Auth.AppRole.SecretRef.Name != "") {
+			v.issuer.GetSpec().Vault.Auth.AppRole.SecretRef.Name != "" ||
+			v.issuer.GetSpec().Vault.Auth.AppRole.Path != "") {
```

Comment:
From what I can tell, `v.issuer.GetSpec().Vault.Auth.AppRole.Path` *might* be blank (as we default it elsewhere in memory). Does this not break approle auth where the Path is not set?

Reply:
Also, just taking a look over the brace *above* this block:

https://github.com/vdesjardins/cert-manager/blob/e3f67ecaab3a8fec4da7d4e86c78d2593cee20c3/pkg/issuer/vault/setup.go#L40-L42

Should this not be using || instead of && for the check?

Either way, can we add some comments to each of these if blocks to make it clear what we're checking for? 😄